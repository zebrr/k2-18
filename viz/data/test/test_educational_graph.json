{
  "_meta": {
    "description": "Educational graph with different node and edge types",
    "purpose": "Testing metrics calculation with various educational relationships",
    "expected_properties": "Concept should have high PageRank, Assessment should test Chunks"
  },
  "nodes": [
    {
      "id": "concept_1",
      "type": "Concept",
      "text": "Sorting",
      "node_offset": 0,
      "definition": "Process of arranging elements in a specific order",
      "difficulty": 1
    },
    {
      "id": "chunk_1",
      "type": "Chunk",
      "text": "Bubble sort is a simple sorting algorithm that repeatedly steps through the list",
      "node_offset": 100,
      "definition": "Introduction to bubble sort algorithm",
      "difficulty": 2
    },
    {
      "id": "chunk_2",
      "type": "Chunk",
      "text": "def bubble_sort(arr): for i in range(len(arr)): for j in range(0, len(arr)-i-1): ...",
      "node_offset": 200,
      "definition": "Implementation of bubble sort in Python",
      "difficulty": 3
    },
    {
      "id": "assessment_1",
      "type": "Assessment",
      "text": "Implement an optimized version of bubble sort that stops early if no swaps occur",
      "node_offset": 300,
      "difficulty": 3
    }
  ],
  "edges": [
    {
      "source": "concept_1",
      "target": "chunk_1",
      "type": "PREREQUISITE",
      "weight": 1.0
    },
    {
      "source": "chunk_1",
      "target": "chunk_2",
      "type": "PREREQUISITE",
      "weight": 1.0
    },
    {
      "source": "chunk_1",
      "target": "concept_1",
      "type": "EXAMPLE_OF",
      "weight": 1.0
    },
    {
      "source": "chunk_2",
      "target": "chunk_1",
      "type": "ELABORATES",
      "weight": 1.0
    },
    {
      "source": "assessment_1",
      "target": "chunk_2",
      "type": "TESTS",
      "weight": 1.0
    }
  ]
}
