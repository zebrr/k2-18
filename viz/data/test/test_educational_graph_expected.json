{
  "_meta": {
    "description": "Educational graph with different node and edge types - EXPECTED METRICS",
    "purpose": "Testing metrics calculation with various educational relationships",
    "expected_properties": "Concept should have high PageRank, Assessment should test Chunks",
    "calculations_verified": "2024-01-15",
    "metrics_computed_with": {
      "pagerank_damping": 0.85,
      "pagerank_max_iter": 100,
      "betweenness_normalized": true,
      "all_weights": 1.0
    }
  },
  "nodes": [
    {
      "id": "concept_1",
      "type": "Concept",
      "text": "Sorting",
      "node_offset": 0,
      "definition": "Process of arranging elements in a specific order",
      "difficulty": 1,
      "degree_in": 1,
      "degree_out": 1,
      "degree_centrality": 0.6667,
      "pagerank": 0.2341,
      "betweenness_centrality": 0.0,
      "out-closeness": 0.4444,
      "component_id": 0,
      "prerequisite_depth": 0,
      "learning_effort": 1,
      "educational_importance": 0.23406
    },
    {
      "id": "chunk_1",
      "type": "Chunk",
      "text": "Bubble sort is a simple sorting algorithm that repeatedly steps through the list",
      "node_offset": 100,
      "definition": "Introduction to bubble sort algorithm",
      "difficulty": 2,
      "degree_in": 2,
      "degree_out": 2,
      "degree_centrality": 1.3333,
      "pagerank": 0.4625,
      "betweenness_centrality": 0.5,
      "out-closeness": 0.6667,
      "component_id": 0,
      "prerequisite_depth": 1,
      "learning_effort": 3,
      "educational_importance": 0.46250
    },
    {
      "id": "chunk_2",
      "type": "Chunk",
      "text": "def bubble_sort(arr): for i in range(len(arr)): for j in range(0, len(arr)-i-1): ...",
      "node_offset": 200,
      "definition": "Implementation of bubble sort in Python",
      "difficulty": 3,
      "degree_in": 2,
      "degree_out": 1,
      "degree_centrality": 1.0,
      "pagerank": 0.2659,
      "betweenness_centrality": 0.3333,
      "out-closeness": 0.4444,
      "component_id": 0,
      "prerequisite_depth": 2,
      "learning_effort": 6,
      "educational_importance": 0.26594
    },
    {
      "id": "assessment_1",
      "type": "Assessment",
      "text": "Implement an optimized version of bubble sort that stops early if no swaps occur",
      "node_offset": 300,
      "difficulty": 3,
      "degree_in": 0,
      "degree_out": 1,
      "degree_centrality": 0.3333,
      "pagerank": 0.0375,
      "betweenness_centrality": 0.0,
      "out-closeness": 0.5,
      "component_id": 0,
      "prerequisite_depth": 0,
      "learning_effort": 3,
      "educational_importance": 0.03750
    }
  ],
  "edges": [
    {
      "source": "concept_1",
      "target": "chunk_1",
      "type": "PREREQUISITE",
      "weight": 1.0,
      "inverse_weight": 1.0
    },
    {
      "source": "chunk_1",
      "target": "chunk_2",
      "type": "PREREQUISITE",
      "weight": 1.0,
      "inverse_weight": 1.0
    },
    {
      "source": "chunk_1",
      "target": "concept_1",
      "type": "EXAMPLE_OF",
      "weight": 1.0,
      "inverse_weight": 1.0
    },
    {
      "source": "chunk_2",
      "target": "chunk_1",
      "type": "ELABORATES",
      "weight": 1.0,
      "inverse_weight": 1.0
    },
    {
      "source": "assessment_1",
      "target": "chunk_2",
      "type": "TESTS",
      "weight": 1.0,
      "inverse_weight": 1.0
    }
  ]
}
